pypeit/edgetrace.py

```diff
@@ -4212,7 +4216,10 @@ def maskdesign_matching(self, debug=False):
                     num += 1
             msgs.info('*' * 92)
 
-        reference_row = self.left_pca.reference_row if self.par['left_right_pca'] else self.pca.reference_row
+        # reference row
+        bpm = self.bitmask.flagged(self.edge_msk, self.bitmask.bad_flags)
+        reference_row = trace.most_common_trace_row(bpm) if self.pcatype is None \
+            else (self.left_pca.reference_row if self.par['left_right_pca'] else self.pca.reference_row)
```

Comment:
Nice.  Don't worry about doing this now, but I wonder if we should instead make `reference_row` an attribute of `EdgeTraceSet`.